lib/rules/no-restricted-imports.js

```diff
@@ -324,6 +339,56 @@ module.exports = {
 			return {};
 		}
 
+		// Build sets of allowed type imports
+		const allowedTypeImportPathNameSet = new Set();
+		for (const restrictedPath of restrictedPaths) {
+			if (
+				typeof restrictedPath === "object" &&
+				restrictedPath.allowTypeImports
+			) {
+				allowedTypeImportPathNameSet.add(restrictedPath.name);
+			}
+		}
+
+		/**
+		 * Check if a type import is allowed for a given path
+		 * @param {string} importSource The path of the import
+		 * @returns {boolean} Whether the type import is allowed
+		 */
+		function isAllowedTypeImportPath(importSource) {
+			return allowedTypeImportPathNameSet.has(importSource);
+		}
```

Comment:
[Bug] I replaced the body of `isAllowedTypeImportPath` with `return true;` and all tests still passed. Which means the logic that fills in `allowedTypeImportPathNameSet` is effectively unused right now.

Reply:
It is used, added a test case in https://github.com/eslint/eslint/pull/19562/commits/f9559eb7ce07895c289c1b7cb5946a3922efca82

We should add it in typescrip-eslint as well 😄

Reply:
Haha, I'm even more excited about deleting the extension rules now...